_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -612,7 +612,9 @@ VAAPIVideoCORE::AllocFrames(
         } else
         {
             // external allocator
-            if (m_bSetExtFrameAlloc && request->Info.FourCC != MFX_FOURCC_P8)
+            if (m_bSetExtFrameAlloc &&
+                request->Info.FourCC != MFX_FOURCC_P8 &&
+                !(request->Type & MFX_MEMTYPE_INTERNAL_FRAME))
             {
                 sts = (*m_FrameAllocator.frameAllocator.Alloc)(m_FrameAllocator.frameAllocator.pthis, &temp_request, response);
```

Comment:
From my understanding, this code block is used only for one case: when decoder at Init needs to access all memId's which had been allocated by app. I'd suggest to add a comment here:

```
// make 'fake' Alloc call to retrieve memId's of surfaces already allocated by app.
sts = (*m_FrameAllocator.frameAllocator.Alloc)(m_FrameAllocator.frameAllocator.pthis, &temp_request, response);

```

Reply:
External block refactored.